src/controls/qml/RemorseTimer.qml

```diff
@@ -0,0 +1,265 @@
+    */
+    signal cancelled()
+
+    Timer {
+        id: timer
+        interval: remorseTimer.interval
+        running: false
+        repeat: false
+        onTriggered: {
+            remorseTimer.visible = false;
+            remorseTimer.opacity = 0;
+            remorseTimer.triggered();
+        }
+    }
+
+    property real arcValue: 100
+    property int countdownSeconds: Math.floor(interval / 1000)
+
+    SegmentedArc {
+        id: countdownArc
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            verticalCenter: parent.verticalCenter
+        }
+        width: Dims.l(22)
+        height: width
+        segmentAmount: remorseTimer.gaugeSegmentAmount
+        inputValue: remorseTimer.arcValue
+        fgColor: "#ffffff"
+        bgColor: Qt.rgba(1, 1, 1, 0.2)
+        start: remorseTimer.gaugeStartDegree
+        endFromStart: remorseTimer.gaugeEndFromStartDegree
+    }
+
+    Label {
+        id: countdownLabel
+        anchors.centerIn: countdownArc
+        font {
+            pixelSize: Dims.l(18)
+            styleName: "SemiBoldCondensed"
+        }
+        color: "#ffffff"
+        text: remorseTimer.countdownSeconds // Direct integer display
+        z: countdownArc.z + 1
+    }
+
+    Label {
+        id: actionLabel
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            bottom: countdownArc.top
+            bottomMargin: Dims.l(1)
+        }
+        font.pixelSize: Dims.l(6)
+        color: "#ffffff"
+        text: action
+    }
+
+    Label {
+        id: cancelLabel
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            top: countdownArc.bottom
+            topMargin: Dims.l(1)
+        }
+        font.pixelSize: Dims.l(6)
+        color: "#ffffff"
+        text: cancelText
+    }
+
+    MouseArea {
+        anchors.fill: parent
+        onClicked: {
+            timer.stop();
+            fadeAnimation.stop();
+            arcAnimation.stop();
+            syncTimer.stop();
+            fadeOutAnimation.start();
+        }
+    }
+
+    NumberAnimation {
+        id: fadeAnimation
+        target: remorseTimer
+        property: "opacity"
+        from: 0
+        to: 1
+        duration: 300
+    }
+
+    NumberAnimation {
+        id: fadeOutAnimation
+        target: remorseTimer
+        property: "opacity"
+        from: 1
+        to: 0
+        duration: 300
+        easing.type: Easing.InOutQuad
+        onStopped: {
+            remorseTimer.visible = false;
+            remorseTimer.cancelled();
+        }
+    }
+
+    NumberAnimation {
+        id: arcAnimation
+        target: remorseTimer
+        property: "arcValue"
+        from: 100
+        to: 0
+        duration: remorseTimer.interval
+        easing.type: Easing.Linear
+    }
+
+    Timer {
+        id: syncTimer
+        interval: remorseTimer.interval / Math.floor(remorseTimer.interval / 1000) // 1000ms for 3000ms
```

Comment:
See comment below, don't use your example value for something the user is probably going to override.

Since these types of comments are obvious, you might as well refrain from using them.

Reply:
Additionally this always returns `1000`. Might as well use that right? 

Example
```
remorseTimer.interval = 3000
interval: 3000 / floor(3000 / 1000) -> 3000 / floor(3) -> 1000

remorseTimer.interval = 5000
interval: 5000 / floor(5000 / 1000) -> 5000 / floor(5) -> 1000
```

Additionally this introduces a bug when `remorseTimer.interval` is less than 1000 as `Math.floor(remorseTimer.interval / 1000)` would yield zero. And division by zero results in errors.